src/Evolution/Systems/Cce/Initialize/NoIncomingRadiation.cpp

```diff
@@ -0,0 +1,186 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Cce/Initialize/NoIncomingRadiation.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/Cce/Initialize/InitializeJ.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/Printf.hpp"
+#include "Time/History.hpp"
+#include "Time/Slab.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Time/TimeSteppers/RungeKutta3.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce {
+namespace InitializeJ {
+namespace detail {
+
+void radial_evolve_psi0_condition(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 2>*> volume_j_id,
+    const SpinWeighted<ComplexDataVector, 2>& boundary_j,
+    const SpinWeighted<ComplexDataVector, 2>& boundary_dr_j,
+    const SpinWeighted<ComplexDataVector, 0>& r, const double radial_step,
+    const size_t l_max, const size_t number_of_radial_points) noexcept {
+  // initialize J and dy J = I with their boundary values
+  // Optimization note: this could be slightly improved with aggregated
+  // allocations via Variables
+  SpinWeighted<ComplexDataVector, 2> radial_evolved_j = boundary_j;
+  SpinWeighted<ComplexDataVector, 2> radial_evolved_i = 0.5 * boundary_dr_j * r;
+
+  TimeSteppers::RungeKutta3 stepper{};
```

Comment:
why not use the boost::numeric::odeint::runge_kutta_dopri5 to do the ode integration? (see `PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Tov.cpp`)

Reply:
Unless I'm misunderstanding the boost utilities (entirely possible, they're rather intricate), the main supported method is as in the `Tov.cpp`, but that method doesn't end up working because the estimate of the l_inf error norm internally in boost assumes that `abs(array[i])` returns a double, which is not the case for an `array<ComplexDataVector, 2>` as I would be inclined to use here.
Unless there's a clear way to use the `odeint` with blaze types that I'm overlooking, It seems like my options would be to either:
- create a `boost::odeint`-compatible 'Algebra' type for working with arrays of datavectors so that those calls don't fail
- do a bunch of internal state-tracking manually and just call the `boost::numeric::odeint::runge_kutta_dopri5` `do_step_impl` functions manually
- use the SpECTRE steppers as is the current version in the PR.